upgrade-notes/backend_20240726_093001.md

```diff
@@ -0,0 +1,53 @@
+#### product files support ([#{3288}](https://github.com/shopsys/shopsys/pull/3288))
+
+-   uploaded files changes
+
+    -   translations introduced for `UploadedFile` entity for support of localized names
+        -   `UploadedFileTranslation` entity was introduced
+        -   `UploadedFile` now extends `AbstractTranslatableEntity`
+        -   `UploadedFileData` properties added
```

Comment:
I do not mind that you mentioned these kinds of changes as well, I just wanted to say we usually do not do that (as they are not BC breaks)

Reply:
on the other hand, I have found a few BC breaking changes that should be mentioned here:
- `FileUpload::getTemporaryFilename()` changed its visibility from protected to public
- removal of `UploadedFile::getEntityName()` and the other methods - however, you mention the removed properties so I believe you can just enhance the note (something like "these properties were moved along with the related functionality into `UploadedFileRelation`"
- `UploadedFile::checkForDelete()` method was removed (without replacement, right?)
- `UploadedFileFactory(+Interface)` methods changes (however, there are default values in the new method arguments, so technically, no BC breaks here :shrug: )
- `ProductDataFactory` constructor change
- `ProductFacade`  constructor change

Reply:
upgrade notes rewritten